### ISSUE TITLE
Problems with ``process_cpu_seconds_total`` metric

### ISSUE THREAD

**Issue by litetex** [coturn/coturn#1560]
The ``process_cpu_seconds_total`` seems to be completely broken:
1. It's a ``gauge`` and not a ``counter``.<br/>Is there some hidden time travel functionallity included? 😄
2. The value is always ``0`` and never increases

**Comment by jelmd** [comment]
If you need a proper and efficient metrics implementation, try https://github.com/jelmd/coturn ;-)

**Comment by eakraly** [comment]
Hi @litetex 
Are you sure you are reporting an issue in the right repo?
coturn does not have `process_cpu_seconds_total` metric - it comes from https://github.com/prometheus/node_exporter

**Comment by jelmd** [comment]
Nope, libprom collects the metric for the process, which is using it - see https://github.com/digitalocean/prometheus-client-c/commit/c57034d196582d99267d027abb52a05a55dc07f6 and https://github.com/jelmd/libprom/blob/main/prom/src/prom_process_collector.c . node exporter collects global system data and the same metric for itself, too ;-)).

**Comment by litetex** [comment]
> Hi @litetex 
>Are you sure you are reporting an issue in the right repo? 
> coturn does not have `process_cpu_seconds_total` metric - it comes from https://github.com/prometheus/node_exporter

Yes this is the correct repo and coturn provides this metric.
I'm scraping these metrics (used docker image is ``coturn/coturn:4-alpine``) and get the following:

```
curl http://localhost:9641/metrics
...
# HELP process_cpu_seconds_total Total user and system CPU time spent in seconds.
# TYPE process_cpu_seconds_total gauge
process_cpu_seconds_total 0
...
```

**Comment by eakraly** [comment]
@litetex @jelmd 

I see now. I suppose that's a bug in https://github.com/digitalocean/prometheus-client-c - it has been abandoned by DO and not getting any fixes unfortunately.
There is no decision yet to switch to some other library...

Obvious suggestion is to not use any of the system metrics reported by coturn itself (unless you find them working)
I myself never used any of coturn `/proc` metrics because I had node_exporter in any case (have to monitor more than just coturn) and because I was not aware of it :)

**Comment by jonesmz** [comment]
FYI, as I've communicated in other issues/PRS, my company leadership has required the use of opentelemetry in our services, so my group will be adopting opentelemetry SDK in our fork of coturn.

We would be happy to upstream that when it is finished, but it will take a while as we have quite a few juggling pins in the air.

As a stopgap we are going to use promethius adapted to open telemetry before switching all the was

**Comment by jelmd** [comment]
> @litetex @jelmd
...
> Obvious suggestion is to not use any of the system metrics reported by coturn itself (unless you find them working)

Actually it just depends on the libprom library used. And IIRC there is no easy/efficient way to reproduce them using a different collector.

> I myself never used any of coturn `/proc` metrics because I had node_exporter in any case (have to monitor more than just coturn) and because I was not aware of it :)

Actually node_exporter does the same: it uses the "Go libprom" to collect process_cpu_* metrics - and thus these are the metrics for the node_exporter process itself (i.e. /proc/$pid/stat etc.). AFAIC node_exporter cannot be used to monitor other processes. So whatever you monitor with node_exporter, the metrics it generates are disjoint from those generated by coturn, even if they have the same name.

**Comment by jelmd** [comment]
> @litetex @jelmd
...
> Obvious suggestion is to not use any of the system metrics reported by coturn itself (unless you find them working)

Actually it just depends on the libprom library used. And IIRC there is no easy/efficient way to reproduce them using a different collector.

> I myself never used any of coturn `/proc` metrics because I had node_exporter in any case (have to monitor more than just coturn) and because I was not aware of it :)

Well, node_exporter does the same: it uses the "Go libprom" to collect `process_cpu_`* metrics - and thus these are the metrics for the node_exporter process itself (i.e. `/proc/$pid/stat` etc.). AFAIC node_exporter cannot be used to monitor other processes. So whatever you monitor with node_exporter, the metrics it generates are disjoint from those generated by coturn, even if they have the same name.
